processor/k8sprocessor/doc.go

```diff
@@ -15,13 +15,27 @@
 // Package k8sprocessor allow automatic tagging of spans, metrics and logs with k8s metadata.
 //
 // The processor automatically discovers k8s resources (pods), extracts metadata from them and adds the
-// extracted metadata to the relevant spans, metrics and logs. The processor use the kubernetes API to discover all pods
-// running in a cluster, keeps a record of their IP addresses and interesting metadata. Upon receiving telemetry data,
-// the processor looks for presence of well-known resource attributes which might contain IP address ("ip",
-// "k8s.pod.ip" for logs, metrics or traces and "host.name" for metrics). If this field is not available, or it
-// does not contain a valid IP address, the processor tries to identify the source IP address of the service
-// that sent the telemetry data.
-// If a match is found, the cached metadata is added to the data as resource attributes.
+// extracted metadata to the relevant spans, metrics and logs. The processor uses the kubernetes API to discover all pods
+// running in a cluster, keeps a record of their IP addresses, pod UIDs and interesting metadata.
+// The rules for associating the source record with specific Pod Metadata are configured via "pod_association" key.
+// It represents a list of rules that are executed in order until the first one is able to do the match.
+// Each rule is specified as a pair of from (representing the rule type) and name (representing the extracted key name).
+// Following rule types are available:
+//   from: "labels" - takes the attribute from resource attributes (the value can contain either IP address or Pod UID)
+//   from: "connection" - takes the IP attribute from connection context (if available) and automatically
+//     associates it with "k8s.pod.ip" attribute
+// Pod association configuration.
+// pod_association:
```

Comment:
What happens if "pod_association" setting is missing or is present but is empty?

Reply:
I think in that case it should continue to work as it does today i.e fallback to incoming connection IP and IP attribute on resource but will let @pmatyjasek-sumo confirm.

Reply:
Can we document that behavior?

Reply:
Ok I'll update it